test/unit/feirari/AutoRewardsDistributor.test.ts

```diff
@@ -0,0 +1,158 @@
+import { expectRevert, balance, getAddresses, getCore } from '../../helpers';
+import { expect } from 'chai';
+import hre, { ethers, artifacts } from 'hardhat';
+import { BigNumber, Signer } from 'ethers';
+import { join } from 'path/posix';
+
+const toBN = ethers.BigNumber.from;
+
+const MockTribalChief = artifacts.readArtifactSync('MockTribalChief');
+const MockRewardsDistributor = artifacts.readArtifactSync('MockRewardsDistributor');
+
+const RewardsDistributorAdmin = artifacts.readArtifactSync('RewardsDistributorAdmin');
+const AutoRewardsDistributor = artifacts.readArtifactSync('AutoRewardsDistributor');
+
+describe('AutoRewardsDistributor', function () {
+  let userAddress: string;
+  let minterAddress: string;
+  let pcvControllerAddress: string;
+  let governorAddress: string;
+  const e18 = '000000000000000000';
+
+  const impersonatedSigners: { [key: string]: Signer } = {};
+
+  before(async () => {
+    const addresses = await getAddresses();
+
+    // add any addresses you want to impersonate here
+    const impersonatedAddresses = [
+      addresses.userAddress,
+      addresses.secondUserAddress,
+      addresses.minterAddress,
+      addresses.pcvControllerAddress,
+      addresses.governorAddress,
+      addresses.genesisGroup,
+      addresses.beneficiaryAddress1,
+      addresses.beneficiaryAddress2,
+      addresses.guardianAddress,
+      addresses.burnerAddress
+    ];
+
+    for (const address of impersonatedAddresses) {
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [address]
+      });
+
+      impersonatedSigners[address] = await ethers.getSigner(address);
+    }
+  });
+
+  beforeEach(async function () {
+    ({ userAddress, pcvControllerAddress, governorAddress, minterAddress } = await getAddresses());
+
+
+    this.poolIndex = 1000;
```

Comment:
Avoid using `this` inside of mocha tests; see https://stackoverflow.com/questions/27358338/mocha-this-in-before-and-beforeeach-hooks

and my comment on Joey's pr as well:

https://github.com/fei-protocol/fei-protocol-core/pull/187#discussion_r719143546

Reply:
This gets redefined before each test so we should be good.

Reply:
I really think we should avoid using `this`. Another reason not to use `this` is that there's no typing on the vars when you do it this way.

Reply:
fixed